public/request/achievement/update-video.php

```diff
@@ -2,18 +2,41 @@
 
 use Illuminate\Support\Arr;
 use Illuminate\Support\Facades\Validator;
+use LegacyApp\Community\Enums\ArticleType;
+use LegacyApp\Platform\Enums\AchievementType;
+use LegacyApp\Platform\Models\Achievement;
 use LegacyApp\Site\Enums\Permissions;
 
-if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::Developer)) {
+if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::JuniorDeveloper)) {
     abort(401);
 }
 
 $input = Validator::validate(Arr::wrap(request()->post()), [
-    'achievement' => 'required|integer',
+    'achievement' => 'required|integer|exists:mysql_legacy.Achievements,ID',
     'video' => 'nullable|url',
 ]);
 
-if (updateAchievementEmbedVideo((int) $input['achievement'], $input['video'])) {
+$achievementId = (int) $input['achievement'];
+$embedUrl = $input['video'];
+
+$achievement = Achievement::find($achievementId);
+
+$currentVideoUrl = $achievement['AssocVideo'];
+
+// Only allow jr. devs to update achievement embed if they are the author and the achievement is not core/official
+if (
+    $permissions == Permissions::JuniorDeveloper &&
+    ($user != $achievement['Author'] || $achievement['Flags'] == AchievementType::OfficialCore)
+) {
+    abort(401);
+}
+
+if (updateAchievementEmbedVideo($achievementId, $embedUrl)) {
+    $embedUrlText = $embedUrl === null ? 'null' : $embedUrl;
+    $auditLog = "$user set this achievement's embed URL to $embedUrlText.";
+
+    addArticleComment('Server', ArticleType::Achievement, $achievementId, $auditLog, $user);
```

Comment:
I'm going to leave this open for @luchaos to review. The achievement audit log is visible to all, even when not logged in. Having old URLs visible might be undesired if we explicitly censure a video for whatever reason (linked video show some form of cheating, or an unintended glitch to make it easier, or something similar). 

Given that it's a dev posting the video, the level of trust might be acceptable.

Maybe we could modify the comment widget to remove URLs from system comments for non-admin users?

Reply:
I too think that the outdated URL should not be present in the audit message publicly.

We don't include values for other audit messages either as far as I'm aware?

ℹ️ A new audit system will be introduced at some point which will include all the changes with values but gated behind certain roles.

Reply:
Re-reading the requirements from QA and DevC, it sounds like it's indeed safe to omit these links and keep the text just saying the embed URL was updated.

I've made this adjustment in https://github.com/RetroAchievements/RAWeb/pull/1526/commits/f1c62c77d3af1025004dcdb0a0634c7de7c8f5ef.

![Screenshot 2023-07-15 at 8 25 39 AM](https://github.com/RetroAchievements/RAWeb/assets/3984985/425f1258-8ab6-47aa-bfa0-607245f8be77)